### PR TITLE
Document custom values

### PR DESCRIPTION
**Pull request by i509VCB**
This documents the custom values Fabric API provides or consumes.

### CODE REVIEW

**Review by Technici4n — APPROVED** [review]

**Comment by liach** [comment]
Should this be moved to individual apis?

**Comment by i509VCB** [comment]
The custom values for indigo, possibly, but I feel that location for the document is more accessible since very few modules consume custom values but all modules provide the module lifecycle.

**Comment by liach** [comment]
The module lifecycle definitely stays here. However module specifics I suggest moving to each module.
In addition, I think this file should be merged with files describing entrypoints offered by each module.

**Comment by i509VCB** [comment]
> describing entrypoints offered by each module

I assume you mean the readme like in the lifecycle events v1 module?

**Comment by liach** [comment]
More specifically, I mean the custom values consumed by the fabric api mods should be placed together with the entrypoints consumed by fabric api mods.

For example, if a `fabric-blah-api-v1` mod consumes both `fabric-blah-api-v1:callback_entry` entrypoint and `fabric-blah-api-v1:custom_logic` values from mod jsons, these had better be documented together because they are both from `fabric.mod.json` of the mods using blah api.

**Comment by i509VCB** [comment]
In your example, would the consumed custom value entries become part of the module readme under a `custom values` category or would that be a secondary markdown file within the module.

Similar question to entrypoints - a category in markdown or it's own file in the module.

**Comment by liach** [comment]
Possibly a separate usage file. The readme imo should be more player oriented. Something like that famous tool attribute api from some youtuber cheating scandal definitely need a player-oriented readme and a separate modder oriented usage.

**Comment by i509VCB** [comment]
I don't see the purpose behind a user readme within each module folder - typically users would be reading the readme at the root of the repository. That readme mentions the user stuff before any mod developer things.

**Comment by liach** [comment]
Then feel free. imo the main thing distinguishing "category in markdown" versus "own file" is the size of the markdown. TLDR!

File: docs/CUSTOM-VALUES.md [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+The use of custom values ranges from enabling/disabling features to extra metadata about fabric api modules.
+
+# Provided by Fabric API
+
+Custom values provided by Fabric API are for use by other mods.
+
+## `fabric-api:module-lifecycle`
+
+Specifies the lifecycle of a fabric api module.
+This may be used by other mods to determine
```
Comment (Daomephsta) [inline docs/CUSTOM-VALUES.md]: Determine what? Seems to be some text missing here.

Comment (i509VCB) [reply docs/CUSTOM-VALUES.md]: Okay now I have determined the answer

**Comment by i509VCB** [comment]
Okay @liach I have moved the indigo custom values into indigo (and wrote a very rudimentary readme for it)

**Review by liach — APPROVED** [review]

File: docs/CUSTOM-VALUES.md [hunk 2]
```diff
@@ -0,0 +1,38 @@
+# Custom values
+
+Fabric API modules may have certain custom values set or check for the presence of specific custom values.
+
+The use of custom values ranges from enabling/disabling features to extra metadata about fabric api modules.
+
+# Provided by Fabric API
```
Comment (Juuxel) [inline docs/CUSTOM-VALUES.md]: Shouldn't this technically be a `##` since `#` is for `# Custom values`? (just a small nitpick)

**Review by Juuxel — APPROVED** [review]

**Review by LambdAurora — APPROVED** [review]
(and I'm backing up @Juuxel's suggestion)

**Review by Mysterious-Dev — APPROVED** [review]

File: docs/CUSTOM-VALUES.md (hunk 2 above)
Comment (haykam821) [reply docs/CUSTOM-VALUES.md]: In that case, the following header should be a level deeper as well.

**Review by modmuss50 — APPROVED** [review]

**Pull request closed without merging**